### PR TITLE
Fixing Parse.ly 3.0 release

### PR DESCRIPTION
**Pull request by pauarge**
## Description

Fixing some errors we saw on Parse.ly 3.0 due to type errors. This PR makes some additional checks on parameters to ensure types are always consistent.

This PR changes files on the wp-parsely source code. We will back port them to the main plugin.

### Fixing Parse.ly 3.0 release

Fixes potential issues on wp-parsely 3.0.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Steps to Test

1. Check out PR.
2. Activate Parse.ly on your site and check it renders correctly.

### CODE REVIEW

File: wp-parsely-3.0/src/class-parsely.php [hunk 1]
```diff
@@ -186,7 +186,7 @@ public function insert_page_header_metadata(): void {
 
 		// Assign default values for LD+JSON
 		// TODO: Mapping of an install's post types to Parse.ly post types (namely page/post).
-		$parsely_page = $this->construct_parsely_metadata( $parsely_options, $post );
+		$parsely_page = $this->construct_parsely_metadata( $parsely_options, get_post( $post ) );
```
Comment (jblz) [inline wp-parsely-3.0/src/class-parsely.php:189]: Another possibility that might be more robust, revert this change:

```suggestion
		$parsely_page = $this->construct_parsely_metadata( $parsely_options, $post );
```
...& up on L#180, something like:

```
$post = get_post( $GLOBALS['post'] );
if ( ! $post instanceof WP_Post ) {
    return;
}
```

**Review by sjinks — DISMISSED** [review]

**Review by acicovic — APPROVED** [review]

File: wp-parsely-3.0/src/class-parsely.php (hunk 1 above)
Comment (jblz) [reply wp-parsely-3.0/src/class-parsely.php]: Yeah, definitely don't override the global :)

**Review by jblz — CHANGES_REQUESTED** [review]
It's deprecated, but we should probably hit `insert_parsely_page`, while we're at it:

https://github.com/Parsely/wp-parsely/blob/4121bb9ebc35a1523e12648e20f09e7008fe875e/src/class-parsely.php#L236-L253

**Review by sjinks — DISMISSED** [review]

**Review by jblz — APPROVED** [review]

File: wp-parsely-3.0/src/class-parsely.php [hunk 2]
```diff
@@ -763,12 +774,17 @@ private function get_top_level_term( int $term_id, string $taxonomy_name ) {
...
 	 */
-	private function get_bottom_level_term( int $post_id, string $taxonomy_name ): string {
-		$terms    = get_the_terms( $post_id, $taxonomy_name );
-		$term_ids = is_array( $terms ) ? wp_list_pluck( $terms, 'term_id' ) : array();
-		$parents  = is_array( $terms ) ? array_filter( wp_list_pluck( $terms, 'parent' ) ) : array();
+	private function get_bottom_level_term( int $post_id, string $taxonomy_name ) {
+		$terms = get_the_terms( $post_id, $taxonomy_name );
+
+		if ( ! is_array( $terms ) ) {
+			return false;
```
Comment (GaryJones) [inline wp-parsely-3.0/src/class-parsely.php:783]: Could we at least keep the type consistency, and return an empty string? What advantage is there to returning boolean `false`?

Comment (pauarge) [reply wp-parsely-3.0/src/class-parsely.php:783]: Actually we want to do this to keep consistency with `get_top_level_term` and the check on line 736.

Comment (GaryJones) [reply wp-parsely-3.0/src/class-parsely.php:783]: The check on line 736 can check for string length instead, or a `=== ''`.

**Review by sjinks — APPROVED** [review]

**Comment by sjinks** [comment]
r2214-stacks

**Comment by sjinks** [comment]
v20220119.2
